graphql_compiler/compiler/expressions.py

```diff
@@ -928,11 +937,22 @@ def to_sql(self, dialect: Any, aliases: AliasesDictType, current_alias: AliasTyp
             # PostgreSQL
             # coalesce to an empty array of the corresponding type
             inner_type = strip_non_null_from_type(self.field_type.of_type)
-            if is_same_type(GraphQLInt, inner_type):
-                sql_array_type = "INT"
-            elif is_same_type(GraphQLString, inner_type):
-                sql_array_type = "VARCHAR"
-            else:
+            graphql_type_to_sql_array_type_dict = {
+                GraphQLInt: "INT",
+                GraphQLString: "VARCHAR",
+                GraphQLID: "UUID",
+                GraphQLBoolean: "BOOL",
+                GraphQLFloat: "FLOAT",
+                GraphQLDate: "DATE",
+                GraphQLDateTime: "TIMESTAMP",
```

Comment:
This also might be problematic down the road. In GraphQL, we don't currently distinguish between datetimes with and without timezones, but Postgres has separate `timestamp` and `timestamptz` types. Not much to be done about it right now, other than maybe opening an issue so we don't forget.

Reply:
Looks like we have #745. Do you think that covers it or should I make a separate issue?

Reply:
It's probably fine, thanks for finding it.